Sources/TripKit/managers/TKGeoMonitorManager.swift

```diff
@@ -0,0 +1,273 @@
+    }
+    
+    return identifiers.contains(identifier)
+  }
+  
+  public func setAlertsEnabled(_ enabled: Bool, for trip: Trip) {
+    setPreference(alertsEnabled: enabled, for: trip)
+    
+    guard enabled
+    else {
+      stopMonitoring()
+      return
+    }
+    
+    monitorRegions(from: trip)
+  }
+  
+  private func setPreference(alertsEnabled: Bool, for trip: Trip) {
+    guard let identifier = identifier(from: trip)
+    else {
+      assertionFailure("Failed to set Identifier, check saveURL's existence")
+      return
+    }
+    
+    let identifiers = alertsEnabled ? [identifier] : []
+    
+    /* LATER: If needs multiple trips at once; make the above a `var` and then:
+    if alertsEnabled {
+      identifiers.append(identifier)
+    } else {
+      identifiers.removeAll { $0 == identifier }
+    }
+    */
+    
+    UserDefaults.shared.set(identifiers, forKey: Keys.alertsEnabled)
+  }
+  
+  /// This contains the list of trip ids that the user had set Alerts on.
+  private func enabledIdentifiers() -> [String]? {
+    return UserDefaults.shared.stringArray(forKey: Keys.alertsEnabled)
+  }
+  
+  private func identifier(from trip: Trip) -> String? {
+    guard let identifier = trip.saveURL?.lastPathComponent
+    else {
+      return nil
+    }
+    return identifier
+  }
+  
+  public func monitorRegions(from trip: Trip) {
+    // Since only one trip can have notifications at a time, there is no need to save other trips, just need to replace the current one.
+    
+    // FIXME: This should use `trip.segment.flatMap(\.geofences)` instead.
+    
+    var nearEndCoordinate: CLLocationCoordinate2D?
+    var regions: [CLCircularRegion] = trip.segments.compactMap { segment in
+      guard let coordinate = segment.start?.coordinate
+      else {
+        return nil
+      }
+      
+      let region: CLCircularRegion =
+        .init(center: coordinate,
+                     radius: Constants.regularRadius,
+                     identifier: segment.order.identifier.rawValue)
+      
+      if segment.order == .end {
+        nearEndCoordinate = coordinate
+      }
+      
+      return region
+    }
+    
+    if let coordinate = nearEndCoordinate {
+      // Add another circular region at the final stop for 500m radius detection
+      let region: CLCircularRegion =
+        .init(center: coordinate,
+              radius: Constants.endRadius,
+              identifier: RegionIdentifier.nearEnd.rawValue)
+      regions.append(region)
+    }
+    
+    monitor(regions: regions)
+  }
+  
+  private func monitor(regions: [CLCircularRegion]) {
+    guard let monitor = geoMonitor
+    else {
+      geoMonitor = .init(enabledKey: Keys.alertsEnabled) { trigger in
+        switch trigger {
+        case .manual:
+          break
+        case .initial:
+          break
+        case .visitMonitoring:
+          break
+        case .regionMonitoring:
+          break
+        case .departedCurrentArea:
+          break
+        }
```

Comment:
We can probably just ignore that parameter. I don't think there's much logic here that could depend on that.

Reply:
I've removed that in https://github.com/skedgo/tripkit-ios/pull/231